### PR TITLE
added FIFO-related functions and read_register()

### PR DESCRIPTION
**Pull request by nebelgrau77**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -270,4 +272,75 @@ where
...
+            /// Is FIFO full and at least one sample has been overwritten?
+            fifo_overrun: match fifo_src & FIFOBitmasks::OVRN {
+                0 => false,
+                _ => true,
+            },
+            /// Is FIFO empty (no unread samples)?
+            fifo_empty: match fifo_level_value {
+                0 => true,
+                _ => false,
+            },
```
Comment (lonesometraveler) [inline src/lib.rs:319]: Do we want to use `match` here? Can we just use boolean expressions instead?

Comment (nebelgrau77) [reply src/lib.rs:319]: OK, will change to boolean expressions as per one of your emails. 

Should I just change it and commit again? Will that work?

Comment (lonesometraveler) [reply src/lib.rs:319]: Yes, please make the changes and push.

Comment (nebelgrau77) [reply src/lib.rs]: OK, done.

**Pull request merged**